### PR TITLE
Move OC.Contactsmenu and OC.Backbone to the server bundle

### PR DESCRIPTION
**Pull request by ChristophWurst**
Three requests less to load :rocket: 

The contactsmenu needs Backbone, OC.Backbone to be specific. This was not set yet in the bundle so I moved oc-backbone and oc-backbone-webdav as well.

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]

**Review by skjnldsv — APPROVED** [review]
Oh my!! :heart_eyes:

**Review by juliusknorr — APPROVED** [review]

**Comment by ChristophWurst** [comment]
CI failure (comments acceptance tests) seems unrelated.

File: core/src/OC/index.js [hunk 1]
```diff
@@ -19,9 +19,21 @@
...
  */
 
+import Backbone from 'backbone';
+
 import Apps from './apps'
+import ContactsMenu from './contactsmenu';
+import {davCall, davSync} from './backbone-webdav';
+
+// Patch Backbone for DAV
+Object.assign(Backbone, {
```
Comment (ChristophWurst) [inline core/src/OC/index.js]: @danxuliu if this causes `Backbone.davCall is not a function` then it's this code that causes it. Maybe we must not use `Object.assign` in this case but just assign the functions as properties of the global Backbone object. I actually tested and in my browser Backbone.davCall was set, so I assume this was working.

Comment (danxuliu) [reply core/src/OC/index.js]: No, `Object.assign` is fine. The problem comes from other two issues. Brace yourselves, I am going to explain them :-P

The first issue comes from Talk. Talk does not namespace the libraries that it uses, so [when Backbone is loaded by Talk it replaces the main `Backbone` variable](https://github.com/nextcloud/spreed/blob/1fbda6ed4b3b7e96c8ffa8ccf1f7e1937c107dc6/js/vendor/backbone/backbone.js#L31).

However, even if that is bad, this is not the problem in itself, because Backbone is namespaced in core; even if the Backbone library is initially loaded in the main `Backbone` variable it is then exported to `OC.Backbone`. Therefore, although when the Talk sidebar is loaded in the Files app it replaces the main `Backbone` variable with its own object `OC.Backbone` is still the object from core; as long as the object from core is used through `OC.Backbone` there is no problem.

The second issue is that, now, the Backbone object to which the DAV functions were added (which is the Backbone object from core) is used through the `Backbone` variable instead of `OC.Backbone` in _backbone-webdav.js_. Before this pull request the DAV functions were wrapped in another function which [set the `Backbone` object to be used in the DAV functions](https://github.com/nextcloud/server/pull/13870/files#diff-052cf83da4708d1c9ce9b1d3574d800cL58); that function [was called with `OC.Backbone`](https://github.com/nextcloud/server/pull/13870/files#diff-052cf83da4708d1c9ce9b1d3574d800cL363) and thus the `Backbone` local variable was bound to that object. With the movement to Webpack the wrapper function was removed, so the references to `Backbone` in the DAV functions no longer refer to a local variable and the object given when initializing them, but to the main `Backbone` variable. That variable is set to the Backbone object of Talk when the Talk sidebar is loaded, so when `OC.Backbone.davSync` is used [it calls `Backbone.davCall`](https://github.com/nextcloud/server/blob/aa2ff51a122432d2aea376ca028179c5b5a70a33/core/src/OC/backbone-webdav.js#L354), which is undefined.

How to fix this then? The easiest way is to replace `Backbone` uses with `OC.Backbone` in _backbone-webdav.js_ ([`model instanceof Backbone.Collection`](https://github.com/nextcloud/server/blob/aa2ff51a122432d2aea376ca028179c5b5a70a33/core/src/OC/backbone-webdav.js#L289) and [`Backbone.davCall(_.extend(params, options), model)`](https://github.com/nextcloud/server/blob/aa2ff51a122432d2aea376ca028179c5b5a70a33/core/src/OC/backbone-webdav.js#L354)).

Given that _backbone-webdav.js_ is written in a generic way that would be a bit ugly, though; a better option would be something like which was done before: make `Backbone` a local variable bound to a specific object instead of refering the main `Backbone` variable. However my Webpack knowledge is almost non-existent, so that should be done by someone else ;-)

Comment (ChristophWurst) [reply core/src/OC/index.js]: Thanks for the details. I'm not sure I understand the problem but could you test my branch `fix/backbone-webpack-noconflict`? I moved the Backbone thingy back to its own module and use the `noConflict` method again. This might have made the difference. Could you please check on that branch?

Just for clarification: is it expected or unexpected to see the dav functions on the global Backbone?

Comment (danxuliu) [reply core/src/OC/index.js]: >  Thanks for the details. I'm not sure I understand the problem

Then I did not explain myself clearly. Sorry :-) Let me try again:
- In the Files app the Backbone object from core can be accessed through the global variables `Backbone` and `OC.Backbone`
- The DAV functions for Backbone are added to the Backbone object from core
- The DAV functions for Backbone use the global variable `Backbone`
- When the Talk sidebar is loaded in the Files app the Backbone object from Talk can be accessed through the global variable `Backbone`, and the Backbone object from core can be accessed through the global variable `OC.Backbone`.
- Result: when the Talk sidebar is loaded in the Files app the DAV functions try to use the Backbone object from Talk, but they should use the Backbone object from core instead.

>  could you test my branch fix/backbone-webpack-noconflict? I moved the Backbone thingy back to its own module and use the noConflict method again. This might have made the difference. Could you please check on that branch?

Yes, unfortunately it does not work.

>  Just for clarification: is it expected or unexpected to see the dav functions on the global Backbone?

It is expected to see the DAV functions on the Backbone object from core; that works as expected in this pull request. The problem lies in using the global variable `Backbone` in _backbone-webdav.js_.

Comment (ChristophWurst) [reply core/src/OC/index.js]: I see. I missed the *global Backbone is used in backbone-webdav* part. Could you check again? I've updated and rebased the branch. It now encapsulates the Backbone instance in the `davSync` function and therefore should be immune to the replacement of the global Backbone instance.

Comment (ChristophWurst) [reply core/src/OC/index.js]: Let's continue this in https://github.com/nextcloud/server/pull/13935
